InnoGotchi.API/Infrastructure.Services/PetConditionService.cs

```diff
@@ -0,0 +1,218 @@
+
+            bool emptyH = true; bool emptyT = true;
+            bool nextT = true, nextH = true;
+
+            while (true)
+            {
+                if (nextH)
+                {
+                    for (; i < hungryRecords.Count(); i++)
+                    {
+                        if (hungryRecords[i].HungerState >= HungerLevel.Normal)
+                        {
+                            startH = hungryRecords[i].ChangesDate;
+                            i++;
+                            emptyH = false;
+                            break;
+                        }
+                    }
+
+                    for (; i < hungryRecords.Count(); i++)
+                    {
+                        if (hungryRecords[i].HungerState < HungerLevel.Normal)
+                        {
+                            endH = hungryRecords[i].ChangesDate;
+                            i++;
+                            break;
+                        }
+                    }
+                }
+
+                if (nextT)
+                {
+                    for (; j < thirstyRecords.Count(); j++)
+                    {
+                        if (thirstyRecords[j].ThirstyState >= ThirstyLevel.Normal)
+                        {
+                            startT = thirstyRecords[j].ChangesDate;
+                            j++;
+                            emptyT = false;
+                            break;
+                        }
+                    }
+
+                    for (; j < thirstyRecords.Count(); j++)
+                    {
+                        if (thirstyRecords[j].ThirstyState < ThirstyLevel.Normal)
+                        {
+                            endT = thirstyRecords[j].ChangesDate;
+                            j++;
+                            break;
+                        }
+                    }
+                }
+
+                if (emptyH == true || emptyT == true)
+                    break;
+
+
+                if (startH < endT && startT < endH)
+                {
+                    var temp_start = startH > startT ? startH : startT;
+                    var temp_end = endH > endT ? endT : endH;
+
+                    count += (temp_end - temp_start).TotalHours;
+
+                    if (temp_end == endT && temp_end == endH)
+                    {
+                        break;
+                    }
+
+                    if (temp_end == endT)
+                    {
+                        startH = temp_end;
+                        nextH = false;
+                        nextT = true;
+                        emptyT = true;
+                    }
+                    else
+                    {
+                        startT = temp_end;
+                        nextT = false;
+                        nextH = true;
+                        emptyH = true;
+                    }
+                }
+                else
+                {
+                    if (startH > endT)
```

Comment:
looks like you can use the ternary condition operator here

Reply:
made